bundles/org.openhab.binding.vwweconnect/src/main/java/org/openhab/binding/vwweconnect/internal/discovery/VWWeConnectDiscoveryService.java

```diff
@@ -0,0 +1,120 @@
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.openhab.binding.vwweconnect.internal.VWWeConnectSession;
+import org.openhab.binding.vwweconnect.internal.handler.VWWeConnectBridgeHandler;
+import org.openhab.binding.vwweconnect.internal.model.BaseVehicle;
+import org.openhab.binding.vwweconnect.internal.model.Vehicle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The discovery service, notified by a listener on the VWWeConnectSession.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class VWWeConnectDiscoveryService extends AbstractDiscoveryService
+        implements DiscoveryService, ThingHandlerService {
+
+    private static final int SEARCH_TIME_SECONDS = 60;
+    private final Logger logger = LoggerFactory.getLogger(VWWeConnectDiscoveryService.class);
+    private @Nullable VWWeConnectBridgeHandler vwWeConnectBridgeHandler;
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+
+    public VWWeConnectDiscoveryService() {
+        super(SUPPORTED_THING_TYPES_UIDS, SEARCH_TIME_SECONDS);
+    }
+
+    @Override
+    public void startScan() {
+        removeOlderResults(getTimestampOfLastScan());
+        logger.debug("VWWeConnectDiscoveryService:startScan");
+
+        if (vwWeConnectBridgeHandler != null) {
+            VWWeConnectSession session = vwWeConnectBridgeHandler.getSession();
+            if (session != null) {
+                HashMap<String, BaseVehicle> vwWeConnectThings = session.getVWWeConnectThings();
+                for (Map.Entry<String, BaseVehicle> entry : vwWeConnectThings.entrySet()) {
+                    BaseVehicle thing = entry.getValue();
+                    if (thing != null) {
+                        logger.info("Thing: {}", thing);
+                        onThingAddedInternal(thing);
+                    }
+                }
+            }
+        }
+    }
+
+    private void onThingAddedInternal(BaseVehicle thing) {
+        logger.debug("VWWeConnectDiscoveryService:OnThingAddedInternal");
+        if (thing instanceof Vehicle) {
+            Vehicle theVehicle = (Vehicle) thing;
+            String vin = theVehicle.getCompleteVehicleJson().getVin();
+            if (vin != null) {
+                ThingUID thingUID = new ThingUID(VEHICLE_THING_TYPE, vin);
+                String label = theVehicle.getCompleteVehicleJson().getName();
+                DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withLabel(label)
+                        .withBridge(bridgeUID).withProperty(VIN, vin).withRepresentationProperty(vin).build();
```

Comment:
The representation property must be the name of the property, not the value.

Reply:
aha :-)

Thanks for the info!

I've now created a new PR #8934 for openHAB3.0, this PR can be closed.